### PR TITLE
Update libX11 to v1.8.7 to fix CVEs 2023-43785, 2023-43786 and 2023-43787

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Update libX11 to v1.8.7 to fix CVEs 2023-43785, 2023-43786 and 2023-43787

###### Change Log  

- Update libX11 to v1.8.7 to fix CVEs 2023-43785, 2023-43786 and 2023-43787
- Remove patches that don't apply to the new version of source code
- Update xorg-x11-proto-devel to 2023.2.
xorg-x11-proto-devel-2019.1 's copy of 'XKBgeom.h' was colliding libX11-1.8.7 's copy. The contents of the files were different. The file has been removed in xorg-x11-proto-devel-2023.2.
The contents of xorg-x11-proto-devel-2019.1 's copy of 'XKBgeom.h' were the same as libX11-1.6.12 's copy.

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-43785
- https://nvd.nist.gov/vuln/detail/CVE-2023-43786
- https://nvd.nist.gov/vuln/detail/CVE-2023-43787

###### Test Methodology

- Local build
- Both packages, and their direct dependencies built successfully in buddy build pipeline: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=442205&view=results

### CODE REVIEW

**Review by mfrw — APPROVED** [review]
LGTM.
Let's please ensure its direct dependencies also build before we merge.

**Comment by neha170** [comment]
> LGTM. Let's please ensure its direct dependencies also build before we merge.

Thanks for pointing this out, Falak. One more package needed to be updated for all the direct dependencies to build.
Both packages and their direct dependencies built successfully here: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=442205&view=results

File: SPECS/xorg-x11-proto-devel/xorg-x11-proto-devel.spec [hunk 1]
```diff
@@ -3,13 +3,13 @@
 # rebuild libX11
 Summary:        X.Org X11 Protocol headers
 Name:           xorg-x11-proto-devel
-Version:        2019.1
-Release:        6%{?dist}
+Version:        2023.2
```
Comment (mfrw) [inline SPECS/xorg-x11-proto-devel/xorg-x11-proto-devel.spec:6]: I hope nothing breaks with this big bump :)

**Review by mfrw — APPROVED** [review]
LGTM

**Pull request merged**